src/main/java/io/vertx/json/schema/OutputUnit.java

```diff
@@ -149,6 +153,74 @@ public OutputUnit addAnnotations(List<OutputUnit> annotations) {
     return this;
   }
 
+  @GenIgnore
+  public OutputUnit setSchemaLocation(String schemaLocation) {
+    this.schemaLocation = schemaLocation;
+    return this;
+  }
+
+  public void checkValidity() throws JsonSchemaValidationException {
+    final URL baseUri = schemaLocation != null ? new URL(schemaLocation) : null;
+
+    final Function<String, String> urlFormatter = fragment -> {
+      if (baseUri == null) {
+        return fragment;
+      }
+      return new URL(fragment, baseUri).href();
+    };
+
+    final String msg = getError();
+
+    // if valid is null, it means that we are a caused by error
+    if (valid == null) {
+      throw new JsonSchemaValidationException(
+        msg == null ? "JsonSchema Validation error" : msg,
+        urlFormatter.apply(getInstanceLocation()),
+        // add some information to the stack trace
+        new StackTraceElement("[" + urlFormatter.apply(getInstanceLocation()) + "]", "<" + getKeyword() + ">", getKeywordLocation(), -1));
+    } else {
+      if (!valid) {
+        // valid is "false" we need to throw an exception
+        if (errors.isEmpty()) {
+          // there are no sub errors, but the validation failed
+          throw new JsonSchemaValidationException(
+            msg == null ? "JsonSchema Validation error" : msg,
+            urlFormatter.apply(getInstanceLocation()),
+            // add some information to the stack trace
+              new StackTraceElement("[" + urlFormatter.apply(getInstanceLocation()) + "]", "<" + getKeyword() + ">", getKeywordLocation(), -1));
+        } else {
+          final JsonSchemaValidationException exception;
+          // there are sub errors, we need to cycle them and create a chain of exceptions
+          JsonSchemaValidationException lastException = null;
+          for (final OutputUnit error : errors) {
+            JsonSchemaValidationException cause;
+            if (lastException == null) {
+              cause = new JsonSchemaValidationException(
+                error.getError(),
+                urlFormatter.apply(error.getInstanceLocation()),
+                // add some information to the stack trace
+                new StackTraceElement("[" + urlFormatter.apply(error.getInstanceLocation()) + "]", "<" + error.getKeyword() + ">", error.getKeywordLocation(), -1));
+            } else {
+              cause = new JsonSchemaValidationException(
+                error.getError(),
+                lastException,
+                urlFormatter.apply(error.getInstanceLocation()),
+                // add some information to the stack trace
+                new StackTraceElement("[" + urlFormatter.apply(error.getInstanceLocation()) + "]", "<" + error.getKeyword() + ">", error.getKeywordLocation(), -1));
+            }
```

Comment:
In my opinion the "if" block is not necessary, simply pass last exception even if it is null. The constructor in the if block does the same.